plugin/vcs/gitlab/gitlab.go

```diff
@@ -896,8 +888,10 @@ func (p *Provider) DeleteWebhook(ctx context.Context, oauthCtx common.OauthConte
 // TODO: The same GitLab API endpoint supports using the HEAD request to only
 // get the file metadata.
 func (p *Provider) readFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, ref string) (*File, error) {
-	url := fmt.Sprintf("%s/projects/%s/repository/files/%s?ref=%s", p.APIURL(instanceURL), repositoryID, url.QueryEscape(filePath), url.QueryEscape(ref))
-	code, body, err := oauth.Get(
+	// Since GitLab of stale version doesn't support get large file content well, we move to use the raw api.
+	// See https://gitlab.com/gitlab-org/gitlab-pages/-/issues/315 for more details.
```

Comment:
@h3n4l - From what I read... I think the linked issue is about when GZIP is enabled, the `Content-Length` header is no longer accurate, especially when the GZIP is performed at the reverse proxy, not GitLab itself. So I think the comment here is a bit misleading, we are in fact abandoning relying on the `Content-Length` header by using the raw API.

Reply:
@h3n4l - just checking in, how are we doing on the followup for this?

Reply:
In #2975.